src/data_source/fetching.rs

```diff
@@ -362,24 +362,18 @@ where
             BackgroundTask::spawn("pruner", async move {
                 for i in 1.. {
                     sleep(cfg.interval()).await;
-                    tracing::info!("pruner woke up for the {i}th time",);
+                    tracing::warn!("pruner woke up for the {i}th time",);
 
                     {
                         let mut storage = fetcher.storage.write().await;
 
                         match storage.storage.prune().await {
                             Ok(Some(height)) => {
                                 storage.pruned_height = Some(height);
-                                if let Err(e) = storage.storage.save_pruned_height(height).await {
-                                    tracing::error!("failed to save pruned height: {e:?}");
-                                    continue;
-                                }
-                                if let Err(e) = storage.commit().await {
-                                    tracing::error!("failed to commit: {e:?}")
-                                }
                             }
                             Ok(None) => (),
                             Err(e) => {
+                                storage.revert().await;
```

Comment:
Put this after the `pruning failed` log so if `revert` does anything weird it will be obvious why